modules/core/src/main/scala/io/chrisdavenport/fuuid/FUUIDGen.scala

```diff
@@ -0,0 +1,45 @@
+package io.chrisdavenport.fuuid
+
+import java.util.UUID
+import cats.implicits._
+import cats.effect.Sync
+
+/**
+  * This trait is an F-algebra representation of the ability to generate FUUID's.
+  * 
+  * At some edge a Sync is required in order to populate the randomness when required.
+  */
+@scala.annotation.implicitNotFound("""Cannot find implicit value for FUUIDGen[${F}].
+Building this implicit value depends on having an implicit
+Sync[${F}] or some equivalent type.""")
+trait FUUIDGen[F[_]]{
+  /**
+   * Creates a Random FUUID
+   */
+  def random: F[FUUID]
+  /**
+   * Creates an FUUID from a String, if it is valid
+   */
+  def fromString(s: String): F[FUUID]
+  /**
+    * Creates an FUUID from a UUID
+    */
+  def fromUUID(uuid: UUID): F[FUUID]
+  /**
+   * Creates a new name-based UUIDv5. NOTE: Not implemented for Scala.js!
+   **/
+  def nameBased(namespace: FUUID, name: String): F[FUUID]
+}
+
+object FUUIDGen {
+  def apply[F[_]](implicit ev: FUUIDGen[F]): FUUIDGen[F] = ev
+
+  implicit def fromSync[F[_]: Sync]: FUUIDGen[F] = new SyncFUUIDGen[F]
```

Comment:
Is this an implicit conversion? I thought those were evil ™️ ... do inform my ignorance.